airbyte-webapp/src/components/ui/ResizablePanels/ResizablePanels.tsx

```diff
@@ -0,0 +1,107 @@
+import { faGripLinesVertical } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import classNames from "classnames";
+import React from "react";
+import { ReflexContainer, ReflexElement, ReflexSplitter } from "react-reflex";
+import { useWindowSize } from "react-use";
+
+import styles from "./ResizablePanels.module.scss";
+
+interface ResizablePanelsProps {
+  leftPanel: {
+    children: React.ReactNode;
+    smallWidthHeader: React.ReactNode;
+    className?: string;
+  };
+  rightPanel: {
+    children: React.ReactNode;
+    smallWidthHeader: React.ReactNode;
+    showPanel: boolean;
+    className?: string;
+    startingFlex?: number;
+  };
+  containerClassName?: string;
+}
+
+interface PanelContainerProps {
+  smallWidthHeader: React.ReactNode;
+  dimensions?: {
+    width: number;
+    height: number;
+  };
+  className?: string;
+}
+
+const LeftPanelContainer: React.FC<React.PropsWithChildren<PanelContainerProps>> = ({
+  children,
+  dimensions,
+  smallWidthHeader,
+  className,
+}) => {
+  const width = dimensions?.width ?? 0;
+  const screenWidth = useWindowSize().width;
+
+  return (
+    <div className={classNames(className, styles.leftPanelContainer)}>
+      {screenWidth > 500 && width < 550 && (
+        <div className={styles.darkOverlay}>
+          <h3>{smallWidthHeader}</h3>
+        </div>
+      )}
+      {children}
+    </div>
+  );
+};
+
+const RightPanelContainer: React.FC<React.PropsWithChildren<PanelContainerProps>> = ({
+  children,
+  dimensions,
+  smallWidthHeader,
+  className,
+}) => {
+  const width = dimensions?.width ?? 0;
+
+  return (
+    <div className={classNames(className, styles.rightPanelContainer)}>
+      {width < 350 && (
+        <div className={styles.lightOverlay}>
+          <h2 className={styles.rotatedHeader}>{smallWidthHeader}</h2>
+        </div>
+      )}
+      {children}
+    </div>
+  );
+};
+// NOTE: ReflexElement will not load its contents if wrapped in an empty jsx tag along with ReflexSplitter.  They must be evaluated/rendered separately.
+
+export const ResizablePanels: React.FC<ResizablePanelsProps> = ({ leftPanel, rightPanel, containerClassName }) => {
+  return (
+    <ReflexContainer className={containerClassName} orientation="vertical">
+      <ReflexElement className={styles.leftPanelStyle} propagateDimensions minSize={150}>
+        <LeftPanelContainer className={leftPanel.className} smallWidthHeader={leftPanel.smallWidthHeader}>
+          {leftPanel.children}
+        </LeftPanelContainer>
+      </ReflexElement>
+      {rightPanel.showPanel && (
+        <ReflexSplitter style={{ border: 0, background: "rgba(255, 165, 0, 0)" }}>
```

Comment:
Ended up moving the styles for this splitter into scss module and just set the background color to `transparent`, since that is what the previous value was doing anyway